lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -193,6 +200,53 @@ defmodule Mix.Compilers.Elixir do
     Mix.shell.info "Compiling #{source} (it's taking more than #{threshold}s)"
   end
 
+  defp warn_for_missing_remote_functions(runtime_remote_calls, source, :module),
+    do: do_warn_for_missing_remote_functions(runtime_remote_calls, source)
+
+  defp warn_for_missing_remote_functions(runtime_remote_calls, source, {:impl, _}),
+    do: do_warn_for_missing_remote_functions(runtime_remote_calls, source)
+
+  defp warn_for_missing_remote_functions(_, _, _),
+    do: :ok
+
+  defp do_warn_for_missing_remote_functions(runtime_remote_calls, source) do
+    Enum.sort_by(runtime_remote_calls, fn {_, _, line} -> line end)
+    |> Enum.each(fn {module, {func, arity} = fas, line} ->
+      if Code.ensure_loaded?(module) do
+        unless fas in exports_for_module(module) do
```

Comment:
This could probably be optimized to only ensure a module is loaded/get its exports once per compilation, but this PR didn't seem to be affecting performance noticeably for me, so I went for simplicity instead.